.github/Makefile

```diff
@@ -0,0 +1,116 @@
+SHELL := /bin/bash
+SUDO := sudo
```

Comment:
Here is a diff between `derrickstolee/git_osx_installers:Makefile` and this version:

```diff
7,25c7
< # TODO Update to 10.15 which is the only SDK that I have installed
< # TODO (in /Applications/Xcode.app/...).
< # TODO We still need to decide what we really want to support.
< 
< # OSX_VERSION := 10.13
< OSX_VERSION := 10.15
< 
< # TODO Remove obsolete paths here.  Add /Library path because it does
< # TODO have the 10.14 SDK (which matches what I remember about Apple
< # TODO supporting the need for developers to support the previous
< # TODO release).
< 
< # TODO Added -L before the SDK_PATH to avoid linker error:
< # TODO    ld: can't map file, errno=22 file '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk' for architecture x86_64
< #
< # TODO Removed -isysroot because I don't understand the issues enough
< # TODO yet and there is a step in the YML to try to remove it.  But
< # TODO this needs more study.
< 
---
> OSX_VERSION := $(shell sw_vers -productVersion)
28,46d9
< # Set a default.
< OSX_NAME := "Catalina"
< 
< ifeq ("$(OSX_VERSION)", "10.13")
< OSX_NAME := "High Sierra"
< endif
< #
< # TODO Maybe add 10.14 here.
< #
< ifeq ("$(OSX_VERSION)", "10.15")
< OSX_NAME := Catalina
< endif
< 
< OSX_CODE := $(shell echo "$(OSX_NAME)" | tr '[:upper:]' '[:lower:]' | tr ' ' '-')
< 
< # TODO Universal means Intel and Apple Silicon.  We need Xcode (I
< # TODO believe) 12.4+ to compile for both.  I have 11.3 on my laptop,
< # TODO so I can't test this.  Xcode 13 requires MacOS 11.2 or better.
< 
49d11
< ARCH_FLAGS_universal := -arch x86_64
55d16
< BAK_FOLDER := $(shell date +%s)
58d18
< GIT_BUILT_FROM_COMMIT :=
60,65c20,21
< DOWNLOAD_LOCATION=https://www.kernel.org/pub/software/scm/git
< 
< XML_CATALOG_FILES=$(shell bin/find-file /usr/local/etc/xml/catalog)
< 
< BUILD_CODE := intel-$(ARCH_CODE)-$(OSX_CODE)
< BUILD_DIR := build/$(BUILD_CODE)
---
> BUILD_CODE := intel-$(ARCH_CODE)
> BUILD_DIR := $(GITHUB_WORKSPACE)/payload
68,69d23
< 
< 
72,85c26
< vars:
< 	# OSX_NAME = $(OSX_NAME)
< 	# OSX_CODE = $(OSX_CODE)
< 	# ARCH = $(ARCH)
< 	# ARCH_CODE = $(ARCH_CODE)
< 	# CFLAGS = $(CFLAGS)
< 	# BUILD_CODE = $(BUILD_CODE)
< 	# PREFIX = $(PREFIX)
< 	# DESTDIR = $(DESTDIR)
< 	# GIT_PREFIX = $(GIT_PREFIX)
< 	# BUILD_DIR = $(BUILD_DIR)
< 	# SDK_PATH = $(SDK_PATH)
< 
< .PHONY: compile download install install-assets install-bin install-man install-subtree image package deploy reinstall setup readme
---
> .PHONY: image pkg payload
89,104d29
< /usr/local/etc/xml/catalog:
< 	brew install docbook
< 
< /usr/local/bin/xmlto:
< 	brew install xmlto
< 
< /usr/local/bin/asciidoc:
< 	brew install asciidoc
< 
< 
< tmp/setup-verified: /usr/local/etc/xml/catalog /usr/local/bin/xmlto /usr/local/bin/asciidoc
< 	grep -q docbook-xsl /usr/local/etc/xml/catalog && exit 0 || (echo "You need docbook-xsl installed to build docs; If it is already installed, uninstall and reinstall it"; brew install docbook-xsl)
< 	touch	$@
< 
< setup: tmp/setup-verified
< 
110,117c35,36
< build/%.tar.gz:
< 	mkdir -p build
< 	curl -L -o build/$*.tar.gz.working "$(DOWNLOAD_LOCATION)/$*.tar.gz"
< 	mv build/$*.tar.gz.working build/$*.tar.gz
< 
< $(BUILD_DIR)/git-$(VERSION)/Makefile: build/git-$(VERSION).tar.gz
< 	mkdir -p $(BUILD_DIR)
< 	tar xzf build/git-$(VERSION).tar.gz -C $(BUILD_DIR)
---
> $(BUILD_DIR)/git-$(VERSION)/osx-built-keychain:
> 	cd $(BUILD_DIR)/git-$(VERSION)/contrib/credential/osxkeychain; $(SUBMAKE) CFLAGS="$(CFLAGS) -g -O2 -Wall"
120c39
< $(BUILD_DIR)/git-$(VERSION)/osx-built: $(BUILD_DIR)/git-$(VERSION)/Makefile
---
> $(BUILD_DIR)/git-$(VERSION)/osx-built:
125,126c44,59
< $(BUILD_DIR)/git-$(VERSION)/osx-built-keychain: $(BUILD_DIR)/git-$(VERSION)/Makefile
< 	cd $(BUILD_DIR)/git-$(VERSION)/contrib/credential/osxkeychain; $(SUBMAKE) CFLAGS="$(CFLAGS) -g -O2 -Wall"
---
> $(BUILD_DIR)/git-$(VERSION)/osx-installed-bin: $(BUILD_DIR)/git-$(VERSION)/osx-built $(BUILD_DIR)/git-$(VERSION)/osx-built-keychain
> 	cd $(BUILD_DIR)/git-$(VERSION); $(SUBMAKE) install
> 	cp $(BUILD_DIR)/git-$(VERSION)/contrib/credential/osxkeychain/git-credential-osxkeychain $(DESTDIR)$(GIT_PREFIX)/bin/git-credential-osxkeychain
> 	mkdir -p $(DESTDIR)$(GIT_PREFIX)/contrib/completion
> 	cp $(BUILD_DIR)/git-$(VERSION)/contrib/completion/git-completion.bash $(DESTDIR)$(GIT_PREFIX)/contrib/completion/
> 	cp $(BUILD_DIR)/git-$(VERSION)/contrib/completion/git-completion.zsh $(DESTDIR)$(GIT_PREFIX)/contrib/completion/
> 	cp $(BUILD_DIR)/git-$(VERSION)/contrib/completion/git-prompt.sh $(DESTDIR)$(GIT_PREFIX)/contrib/completion/
> 	# This is needed for Git-Gui, GitK
> 	mkdir -p $(DESTDIR)$(GIT_PREFIX)/lib/perl5/site_perl
> 	[ ! -f $(DESTDIR)$(GIT_PREFIX)/lib/perl5/site_perl/Error.pm ] && cp $(BUILD_DIR)/git-$(VERSION)/perl/private-Error.pm $(DESTDIR)$(GIT_PREFIX)/lib/perl5/site_perl/Error.pm || echo done
> 	ruby scripts/symlink-git-hardlinks.rb $(DESTDIR)
> 	touch $@
> 
> $(BUILD_DIR)/git-$(VERSION)/osx-installed-man: $(BUILD_DIR)/git-$(VERSION)/osx-installed-bin
> 	mkdir -p $(DESTDIR)$(GIT_PREFIX)/share/man
> 	cp -R $(GITHUB_WORKSPACE)/manpages/ $(DESTDIR)$(GIT_PREFIX)/share/man
129c62
< $(BUILD_DIR)/git-$(VERSION)/osx-built-subtree: $(BUILD_DIR)/git-$(VERSION)/Makefile | setup
---
> $(BUILD_DIR)/git-$(VERSION)/osx-built-subtree:
149,166c82
< $(BUILD_DIR)/git-$(VERSION)/osx-installed-bin: $(BUILD_DIR)/git-$(VERSION)/osx-built $(BUILD_DIR)/git-$(VERSION)/osx-built-keychain $(DESTDIR)$(GIT_PREFIX)/VERSION-$(VERSION)-$(BUILD_CODE)
< 	cd $(BUILD_DIR)/git-$(VERSION); $(SUBMAKE) install
< 	cp $(BUILD_DIR)/git-$(VERSION)/contrib/credential/osxkeychain/git-credential-osxkeychain $(DESTDIR)$(GIT_PREFIX)/bin/git-credential-osxkeychain
< 	mkdir -p $(DESTDIR)$(GIT_PREFIX)/contrib/completion
< 	cp $(BUILD_DIR)/git-$(VERSION)/contrib/completion/git-completion.bash $(DESTDIR)$(GIT_PREFIX)/contrib/completion/
< 	cp $(BUILD_DIR)/git-$(VERSION)/contrib/completion/git-completion.zsh $(DESTDIR)$(GIT_PREFIX)/contrib/completion/
< 	cp $(BUILD_DIR)/git-$(VERSION)/contrib/completion/git-prompt.sh $(DESTDIR)$(GIT_PREFIX)/contrib/completion/
< 	# This is needed for Git-Gui, GitK
< 	mkdir -p $(DESTDIR)$(GIT_PREFIX)/lib/perl5/site_perl
< 	[ ! -f $(DESTDIR)$(GIT_PREFIX)/lib/perl5/site_perl/Error.pm ] && cp $(BUILD_DIR)/git-$(VERSION)/perl/private-Error.pm $(DESTDIR)$(GIT_PREFIX)/lib/perl5/site_perl/Error.pm || echo done
< 	ruby UserScripts/symlink_git_hardlinks.rb $(DESTDIR)
< 	touch $@
< 
< $(BUILD_DIR)/git-$(VERSION)/osx-installed-man: build/git-manpages-$(VERSION).tar.gz $(BUILD_DIR)/git-$(VERSION)/osx-installed-bin
< 	tar xzfo build/git-manpages-$(VERSION).tar.gz -C $(DESTDIR)$(GIT_PREFIX)/share/man
< 	touch $@
< 
< $(BUILD_DIR)/git-$(VERSION)/osx-installed: $(BUILD_DIR)/git-$(VERSION)/osx-installed-bin $(BUILD_DIR)/git-$(VERSION)/osx-installed-man $(BUILD_DIR)/git-$(VERSION)/osx-installed-assets $(BUILD_DIR)/git-$(VERSION)/osx-installed-subtree
---
> $(BUILD_DIR)/git-$(VERSION)/osx-installed: $(DESTDIR)$(GIT_PREFIX)/VERSION-$(VERSION)-$(BUILD_CODE) $(BUILD_DIR)/git-$(VERSION)/osx-installed-man $(BUILD_DIR)/git-$(VERSION)/osx-installed-assets $(BUILD_DIR)/git-$(VERSION)/osx-installed-subtree
182,183c98
< 
< disk-image/VERSION-$(VERSION)-$(ARCH_CODE)-$(OSX_CODE):
---
> disk-image/VERSION-$(VERSION)-$(ARCH_CODE):
184a100
> 	mkdir disk-image
187,188c103,104
< disk-image/git-$(VERSION)-$(BUILD_CODE).pkg: disk-image/VERSION-$(VERSION)-$(ARCH_CODE)-$(OSX_CODE) $(DESTDIR)$(GIT_PREFIX)/VERSION-$(VERSION)-$(BUILD_CODE) $(BUILD_DIR)/git-$(VERSION)/osx-installed $(BUILD_DIR)/git-$(VERSION)/osx-built-assert-$(ARCH_CODE)
< 	pkgbuild --identifier com.git.pkg --version $(VERSION) --root $(DESTDIR)$(PREFIX) --scripts scripts --install-location $(PREFIX) --component-plist ./git-components.plist disk-image/git-$(VERSION)-$(BUILD_CODE).pkg
---
> disk-image/git-$(VERSION)-$(BUILD_CODE).pkg: disk-image/VERSION-$(VERSION)-$(ARCH_CODE)
> 	pkgbuild --identifier com.git.pkg --version $(VERSION) --root ../../build_artifacts$(PREFIX) --scripts scripts --install-location $(PREFIX) --component-plist ./assets/git-components.plist disk-image/git-$(VERSION)-$(BUILD_CODE).pkg
190c106
< git-%-$(BUILD_CODE).dmg: disk-image/git-%-$(BUILD_CODE).pkg
---
> git-%-$(BUILD_CODE).dmg:
192c108
< 	hdiutil create git-$(VERSION)-$(BUILD_CODE).uncompressed.dmg -fs HFS+ -srcfolder disk-image -volname "Git $(VERSION) $(OSX_NAME) Intel $(ARCH)" -ov
---
> 	hdiutil create git-$(VERSION)-$(BUILD_CODE).uncompressed.dmg -fs HFS+ -srcfolder ../../disk-image -volname "Git $(VERSION) Intel $(ARCH)" -ov
196,209c112
< tmp/deployed-%-$(BUILD_CODE): git-%-$(BUILD_CODE).dmg
< 	mkdir -p tmp
< 	scp git-$(VERSION)-$(BUILD_CODE).dmg timcharper@frs.sourceforge.net:/home/pfs/project/git-osx-installer | tee $@.working
< 	mv $@.working $@
< 
< package: disk-image/git-$(VERSION)-$(BUILD_CODE).pkg
< install-assets: $(BUILD_DIR)/git-$(VERSION)/osx-installed-assets
< install-bin: $(BUILD_DIR)/git-$(VERSION)/osx-installed-bin
< install-man: $(BUILD_DIR)/git-$(VERSION)/osx-installed-man
< install-subtree: $(BUILD_DIR)/git-$(VERSION)/osx-installed-subtree
< 
< install: $(BUILD_DIR)/git-$(VERSION)/osx-installed
< 
< download: build/git-$(VERSION).tar.gz build/git-manpages-$(VERSION).tar.gz
---
> payload: $(BUILD_DIR)/git-$(VERSION)/osx-installed $(BUILD_DIR)/git-$(VERSION)/osx-built-assert-$(ARCH_CODE)
211,231c114
< compile: $(BUILD_DIR)/git-$(VERSION)/osx-built $(BUILD_DIR)/git-$(VERSION)/osx-built-keychain $(BUILD_DIR)/git-$(VERSION)/osx-built-subtree
< 
< deploy: tmp/deployed-$(VERSION)-$(BUILD_CODE)
< 
< tmp/deployed-readme: README.md
< 	scp README.md timcharper@frs.sourceforge.net:/home/pfs/project/git-osx-installer | tee $@.working
< 	mv $@.working $@
< 
< readme: tmp/deployed-readme
< 
< 
< clean:
< 	$(SUDO) rm -rf $(BUILD_DIR)/git-$(VERSION)/osx-* $(DESTDIR)
< 	[ -d $(BUILD_DIR)/git-$(VERSION) ] && cd $(BUILD_DIR)/git-$(VERSION) && $(SUBMAKE) clean || echo done
< 	[ -d $(BUILD_DIR)/git-$(VERSION)/contrib/credential/osxkeychain ] && cd $(BUILD_DIR)/git-$(VERSION)/contrib/credential/osxkeychain && $(SUBMAKE) clean || echo done
< 	[ -d $(BUILD_DIR)/git-$(VERSION)/contrib/subtree ] && cd $(BUILD_DIR)/git-$(VERSION)/contrib/subtree && $(SUBMAKE) clean || echo done
< 
< reinstall:
< 	$(SUDO) rm -rf /usr/local/git/VERSION-*
< 	rm -f $(BUILD_DIR)/git-$(VERSION)/osx-installed*
< 	$(SUBMAKE) install
---
> pkg: disk-image/git-$(VERSION)-$(BUILD_CODE).pkg
```

Reply:
Looking at this diff, I definitely see a lot of "build magic" but what is clear is that your version is much simpler and more flexible than what existed before. Thanks for doing this!